### PR TITLE
Fix gosec G307 in tar.go

### PR DESCRIPTION
**Pull request by m14815**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
fix gosec G307 in tar.go
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
we had similar fixes submitted for defer closing a file, but arguably this is not needed.

**Comment by xmudrii** [comment]
@dims @spiffxp @BenTheElder Can you please take a look at this PR?

**Comment by BenTheElder** [comment]
cc @johnSchnake

File: cluster/images/conformance/go-runner/tar.go [hunk 1]
```diff
@@ -39,7 +40,11 @@ func tarDir(dir, outpath string) error {
 	if err != nil {
 		return errors.Wrapf(err, "creating tarball %v", outpath)
 	}
-	defer outfile.Close()
+	defer func() {
+		if err := outfile.Close(); err != nil {
+			log.Printf("Error occures while closing file, Error %v", err)
```
Comment (BenTheElder) [inline cluster/images/conformance/go-runner/tar.go:45]: nit: s/occures/occured

File: cluster/images/conformance/go-runner/tar.go [hunk 2]
```diff
@@ -75,7 +80,11 @@ func tarDir(dir, outpath string) error {
 		if err != nil {
 			return errors.Wrapf(err, "opening file %v for writing into tarball", file)
 		}
-		defer f.Close()
+		defer func() {
+			if err := f.Close(); err != nil {
+				log.Printf("Error occures while closing file stream, Error: %v", err)
```
Comment (BenTheElder) [inline cluster/images/conformance/go-runner/tar.go:85]: same nit here

**Pull request closed without merging**